dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/tracker/export/trackedentity/TrackedEntitiesExportControllerPostgresTest.java

```diff
@@ -0,0 +1,349 @@
+import org.springframework.core.io.ClassPathResource;
+
+class TrackedEntitiesExportControllerPostgresTest extends DhisControllerIntegrationTest {
+
+  @Autowired private ObjectBundleService objectBundleService;
+
+  @Autowired private ObjectBundleValidationService objectBundleValidationService;
+
+  @Autowired private RenderService _renderService;
+
+  @Autowired private TrackerImportService trackerImportService;
+
+  @Autowired private TrackedEntityService trackedEntityService;
+
+  @Autowired private ProgramService programService;
+
+  private Program program;
+
+  private TrackedEntity trackedEntity;
+
+  @BeforeEach
+  void setUp() throws IOException {
+    this.renderService = _renderService;
+    setUpMetadata("tracker/simple_metadata.json");
+
+    TrackerImportParams params = TrackerImportParams.builder().userId(superUser.getUid()).build();
+    assertNoDataErrors(
+        trackerImportService.importTracker(params, fromJson("tracker/single_tei.json")));
+
+    trackedEntity = trackedEntityService.getTrackedEntity("IOR1AXXl24H");
+    program = programService.getProgram("BFcipDERJnf");
+
+    JsonWebMessage importResponse =
+        POST("/tracker?async=false&importStrategy=UPDATE", createJsonPayload(2))
+            .content(HttpStatus.OK)
+            .as(JsonWebMessage.class);
+    assertEquals(HttpStatus.OK.toString(), importResponse.getStatus());
+
+    importResponse =
+        POST("/tracker?async=false&importStrategy=UPDATE", createJsonPayload(3))
+            .content(HttpStatus.OK)
+            .as(JsonWebMessage.class);
+    assertEquals(HttpStatus.OK.toString(), importResponse.getStatus());
+
+    importResponse =
+        POST("/tracker?async=false&importStrategy=UPDATE", createJsonPayload(4))
+            .content(HttpStatus.OK)
+            .as(JsonWebMessage.class);
+    assertEquals(HttpStatus.OK.toString(), importResponse.getStatus());
+  }
+
+  @Test
+  void shouldGetTrackedEntityAttributeChangeLogWhenValueCreatedAndThenUpdated() {
+    JsonList<JsonTrackedEntityChangeLog> changeLogs =
+        GET(
+                "/tracker/trackedEntities/{id}/changeLogs?program={programUid}",
+                trackedEntity.getUid(),
+                program.getUid())
+            .content(HttpStatus.OK)
+            .getList("changeLogs", JsonTrackedEntityChangeLog.class);
+
+    JsonTrackedEntityChangeLog changeLog = changeLogs.get(2);
+    JsonUser createdBy = changeLog.getCreatedBy();
+    JsonAttributeValue attributeChange = changeLog.getChange().getAttributeValue();
+    UserDetails currentUser = CurrentUserUtil.getCurrentUserDetails();
+
+    assertAll(
+        () -> assertEquals(currentUser.getUid(), createdBy.getUid()),
+        () -> assertEquals(currentUser.getUsername(), createdBy.getUsername()),
+        () -> assertEquals(currentUser.getFirstName(), createdBy.getFirstName()),
+        () -> assertEquals(currentUser.getSurname(), createdBy.getSurname()),
+        () -> assertEquals("CREATE", changeLog.getType()),
```

Comment:
Please use the same style as in EventsExportControllerPostgresTest
assertUpdate, assertDelete, ...

You can extract them in order to reuse them.

Reply:
I'm refactoring this in the next PR, where I'm dealing with sorting